core/src/main/java/org/apache/calcite/rel/type/RelDataTypeSystemImpl.java

```diff
@@ -357,6 +358,27 @@ && getDefaultPrecision(typeName) != RelDataType.PRECISION_NOT_SPECIFIED) {
                   argumentType.isNullable());
         }
       }
+      // For DECIMAL using default Precision and Scale,
```

Comment:
as discussed in the JIRA, the right approach is to actually create a new type system that overrides this method rather than change RelDataTypeSystemImpl.